### ISSUE TITLE
[win] i-Builds repeatedly DNF I20241211-0310

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform.ui#2620]
https://ci.eclipse.org/releng/job/AutomatedTests/job/ep435I-unit-win32-x86_64-java17/
![image](https://github.com/user-attachments/assets/0a23d955-005c-47a3-9773-0eb9a0a58153)

**Comment by jukzi** [comment]
cc @BeckerWdf got time to help?

**Comment by BeckerWdf** [comment]
Sorry. From your description I don't know what to do.

**Comment by jukzi** [comment]
> Sorry. From your description I don't know what to do.

Probably nobody does -> investigate what exactly went wrong, find out why, solve.

**Comment by BeckerWdf** [comment]
What does "DNF" mean?

**Comment by jukzi** [comment]
as reported in  https://download.eclipse.org/eclipse/downloads/drops4/I20241211-0310/testResults.php

**Comment by jukzi** [comment]
SWT had 1 error then platform.ui DNF:
![image](https://github.com/user-attachments/assets/44288feb-229e-4505-b6c2-5709175570ed)

**Comment by HeikoKlare** [comment]
I just commented on that issue here: https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2648#issuecomment-2536879855

It was very likely caused by merging this PR yesterday: https://github.com/eclipse-platform/eclipse.platform.swt/pull/1637

Sorry that no one of us responded on the issues so far. I just pinged the colleagues to have a look. I am currently out of office and thus, unfortunately, cannot help except for pointing to the cause. At worst, we could revert that PR if there is no quick solution.

**Comment by fedejeanne** [comment]
I'm looking into it as we speak. The error log mentions the folder **EBWebView** so I will look closer to when the userdata directory for Edge was introduced in (https://github.com/eclipse-platform/eclipse.platform.swt/pull/1548 ).

```
Unable to delete file C:\Users\genie.releng\workspace\AutomatedTests\ep435I-unit-win32-x86_64-java17\workarea\I20241211-1800\eclipse-testing\test-eclipse\eclipse\update_configurator_folder\.metadata\.plugins\org.eclipse.swt\EBWebView\lockfile
```

@amartya4256 _ping!_

**Comment by sratz** [comment]
I think this is a timing issue.
The `eclipse.platform.releng.aggregator\production\testScripts\configuration\sdk.tests\testScripts\test.xml` script starts many runtimes in quick succession:
- Run tests
- Delete workspace
- Run tests
- Delete workspace
- ...

Now, with the Edge browser, we spawn `msedgewebview2.exe` child processes.
When shutting down the runtime, it looks like those processes linger around for a little while, holding on to a lock in the workspace.

What I am not quite sure about: Why does this happen after the `org.eclipse.update.configurator.tests.AutomatedConfiguratorSuite` but not after the other ones before it?

**Comment by sratz** [comment]
> junit tests are responsible to wait until their resources have been cleaned. Especially a (the) failed JUnit test should not leak such resources. please adapt the code as necessary.

Not sure if the individual tests can (or should) influence this. They do not spawn those WebView2 processes explicitly.

If anything, there is some `Browser` instance that is not disposed correctly. I'll have a look at the `org.eclipse.update.configurator.tests.AutomatedConfiguratorSuite` test suite to see if I can find anything suspicious.

**Comment by fedejeanne** [comment]
> I think this is a timing issue. The `eclipse.platform.releng.aggregator\production\testScripts\configuration\sdk.tests\testScripts\test.xml` script starts many runtimes in quick succession:
> 
>     * Run tests
> 
>     * Delete workspace
> 
>     * Run tests
> 
>     * Delete workspace
> 
>     * ...
> 
> 
> Now, with the Edge browser, we spawn `msedgewebview2.exe` child processes. When shutting down the runtime, it looks like those processes linger around for a little while, holding on to a lock in the workspace.

This was exactly what we thought and @amartya4256 is trying to validate this theory locally as we speak.

> What I am not quite sure about: Why does this happen after the `org.eclipse.update.configurator.tests.AutomatedConfiguratorSuite` but not after the other ones before it?

Probably this configuration:
https://github.com/eclipse-platform/eclipse.platform/blob/master/update/org.eclipse.update.configurator.tests/test.xml#L37-L47

**Comment by sratz** [comment]
`AutomatedConfiguratorSuite` only contains a single test class/method, has no dependency to SWT whatsoever.
Why is an Edge initialized in the first place?

**Comment by jukzi** [comment]
If there is no solution in sight today i suggest a revert until problem is understood.

**Comment by fedejeanne** [comment]
> `AutomatedConfiguratorSuite` only contains a single test class/method, has no dependency to SWT whatsoever. Why is an Edge initialized in the first place?

The _Intro_ view... for whatever reason.

![image](https://github.com/user-attachments/assets/7c5e781a-aa3f-4d2b-b4be-78e79d18bbce)

If put a breakpoint in `org.eclipse.swt.browser.Edge.create(Composite, int)` and debug the test, you'll see it.

**Comment by akurtakov** [comment]
Intro view is a web page so it loads the default browser engine on each platform in order to show itself (render the webpage).

**Comment by sratz** [comment]
True, but according to the build log we have 47 other rests suite run before `update.configurator.tests`.

I'd expect the intro page to come up in all of them. What makes the update configurator one special?

**Comment by sratz** [comment]
Cannot reproduce locally:

```
java -jar plugins\org.eclipse.equinox.launcher_1.6.900.v20240613-2009.jar -application org.eclipse.test.uitestapplication -data .ws -testPluginName org.eclipse.update.configurator.tests -className org.eclipse.update.configurator.tests.AutomatedConfiguratorSuite -os win32 -ws win32 -arch x86_64 -consolelog
rmdir /s /q .ws
```
is able to delete the whole workspace without problem :(
